lib/routes/account.js

```diff
@@ -1043,16 +1128,18 @@ module.exports = function (
         log.begin('Account.RecoveryEmailResend', request)
         var sessionToken = request.auth.credentials
         var service = request.payload.service || request.query.service
-        if (sessionToken.emailVerified ||
-            Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
-          return reply({})
-        }
-        customs.check(
-          request.app.clientAddress,
-          sessionToken.email,
-          'recoveryEmailResendCode')
-          .then(
-            mailer.sendVerifyCode.bind(
+
+        if (!sessionToken.emailVerified) {
+          // Send email verification
+          if (Date.now() - sessionToken.verifierSetAt < config.smtp.resendBlackoutPeriod) {
+            return reply({})
+          }
+
+          return customs.check(
+            request.app.clientAddress,
+            sessionToken.email,
+            'recoveryEmailResendCode')
+            .then(mailer.sendVerifyCode.bind(
               mailer,
               sessionToken,
               sessionToken.emailCode,
```

Comment:
Sorry for the confusion and the typo, indeed I meant "resend email" not "reset" :-/

> where is "you must verify your account" [...]? In the content server, or the browser?

In both places IIUC - the browser will have its little yellow "you account is unverified" warning marker thing, and the content-server will transition to show the "please confirm you email" screen.  Doing a "resend email" from either of these places, will result in the "verify your account" email being resent.

With the current code here, that email would contain the verification code for the *original* session.  My suggestion is that it should contain the verification code the the *current* session.